### PR TITLE
Added widgets and initial conversion logic

### PR DESCRIPTION
**Pull request by sohel-rp**
# Description

These changes include an initial JSON schema to map conversion logic & the following widgets,

- Styled input & textarea
- Slider
- Styled dropdown
- Checkbox with tippy
- Form with box wrapper

Fixes #https://github.com/devtron-labs/devtron/issues/1869 & #https://github.com/devtron-labs/devtron/issues/1886

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All tests have been performed manually using dummy schema

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/common/formFields/Widgets.tsx [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+        <div className="info-label flex left align-center fs-11 fw-4 mt-4">
+            <span className="info-label-icon">
+                <Info className="icon-dim-16" />
+            </span>
+            <span className="ml-4 cn-7">{description}</span>
+        </div>
+    )
+}
+
+interface StyledFieldPropsType {
```
Comment (vivek-devtron) [inline src/components/common/formFields/Widgets.tsx:47]: Can we extract all the interfaces in separate type file?

File: src/components/common/formFields/Widgets.tsx [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+        <StyledField
+            title={title}
+            description={description}
+            rootClassName={rootClassName}
+            isRequired={isRequired}
+            errorMessage={errorMessage}
+        >
+            <input
+                placeholder={placeholder && placeholder}
+                name={title.replace(/\s/g, '').toLowerCase()}
```
Comment (vivek-devtron) [inline src/components/common/formFields/Widgets.tsx:115]: Can we replace space with _ or something for better readability?

File: src/components/common/formFields/Checkbox.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ export enum CHECKBOX_VALUE {
     CHECKED = 'CHECKED',
     INTERMEDIATE = 'INTERMEDIATE',
 }
-interface CheckboxProps {
+export interface CheckboxProps {
```
Comment (shivani170) [inline src/components/common/formFields/Checkbox.tsx:7]: Can we abstract it out in type file?

File: src/components/common/formFields/Widgets.scss [hunk 4]
```diff
@@ -0,0 +1,89 @@
+.styled-field.form__row {
+    input.form__input:not(.form__textarea) {
+        height: 32px;
+    }
+
+    .info-label-icon,
+    .error-label-icon {
```
Comment (shivani170) [inline src/components/common/formFields/Widgets.scss:7]: Aren't we able to use icon-dim-16 here?

File: src/components/common/formFields/Widgets.scss [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        width: 16px;
+        height: 16px;
```
Comment (shivani170) [inline src/components/common/formFields/Widgets.scss:9]: Can we minimize the size of CSS using inline classnames?

File: src/components/common/formFields/Widgets.tsx [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+    onInput?: (e) => void
+    hideValueInput?: boolean
+}
+
+const sliderRangeUpdate = (e, value: number, min: number, max: number) => {
+    const newValue = ((value - min) / (max - min)) * 100
+    e.style.background = `linear-gradient(to right, var(--B500) 0%, var(--B500) ${newValue}%, var(--N200) ${newValue}%, var(--N200) 100%)`
+}
+
+export const Slider = ({
```
Comment (shivani170) [inline src/components/common/formFields/Widgets.tsx:179]: this component is going way more lengthy. Can we break it down in sub components?

File: src/components/common/formFields/Widgets.tsx [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+                if (!hideValueInput) {
+                    setSliderInputValue(e.target.value)
+                }
+                setSliderValue(value)
+                onInput(value)
+                sliderRangeUpdate(sliderRef.current, value, minLength, maxLength)
+            }
+        } else {
+            if (!hideValueInput) {
+                setSliderInputValue(`${defaultValue ?? minLength}`)
```
Comment (vivek-devtron) [inline src/components/common/formFields/Widgets.tsx:215]: can't we use it directly like defaultValue ?? minLength?

File: src/components/v2/values/chartValuesDiff/ChartValuesView.scss [hunk 8]
```diff
@@ -376,12 +391,12 @@
                     }
                 }
             }
+        }
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.scss:394]: it seems extra. please verify

File: src/components/common/formFields/Widgets.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/common/formFields/Widgets.tsx]: Done

File: src/components/common/formFields/Widgets.tsx (hunk 2 above)
Comment (sohel-rp) [reply src/components/common/formFields/Widgets.tsx]: Done

File: src/components/common/formFields/Checkbox.tsx (hunk 3 above)
Comment (sohel-rp) [reply src/components/common/formFields/Checkbox.tsx:7]: Yup. It'll be covered in one of the standardization tasks. As of now just exported it.

File: src/components/common/formFields/Widgets.scss (hunk 4 above)
Comment (sohel-rp) [reply src/components/common/formFields/Widgets.scss]: removed the width & height from here

File: src/components/common/formFields/Widgets.tsx (hunk 6 above)
Comment (sohel-rp) [reply src/components/common/formFields/Widgets.tsx]: As discussed, this a common component containing different widgets as small comps.

File: src/components/v2/values/chartValuesDiff/ChartValuesView.scss (hunk 8 above)
Comment (sohel-rp) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.scss:394]: Ohh no. I have moved `.chart-values-view__update-cta` outside hence it's showing like this.

**Review by vivek-devtron — APPROVED** [review]

**Review by shivani170 — APPROVED** [review]

**Pull request merged**